syntax = "proto3";
//python -m grpc_tools.protoc --proto_path=./proto/ ./proto/service.proto --python_out=. --grpc_python_out=.

message Request {
    bytes waveform = 1;
    string text = 2;
}

message Response {
    repeated Character scored_character = 1;
}

service Service {
    rpc Recognize(Request) returns (Response);
};

message Character {
    string reference_pinyin = 1;
    repeated string reference_phone = 2;
    repeated string recognized_phone = 3;
    string recognized_pinyin = 4;
    repeated float score_phone = 5;
    string xmin = 6;
    string xmax = 7;
    string summary = 8;
}